source/slang/preprocessor.cpp

```diff
@@ -1875,9 +1885,16 @@ SLANG_PRAGMA_DIRECTIVE_CALLBACK(handlePragmaOnceDirective)
     // trivial cases of the "same" path.
     //
     auto directiveLoc = GetDirectiveLoc(context);
-    auto pathIssuedFrom = context->preprocessor->translationUnit->compileRequest->getSourceManager()->getPath(directiveLoc, SourceLocType::Actual);
+    auto issuedFromPathInfo = context->preprocessor->translationUnit->compileRequest->getSourceManager()->getPathInfo(directiveLoc, SourceLocType::Actual);
+
+    // Must have a canonical path for a #pragma once to work
+    if (!issuedFromPathInfo.hasCanonicalPath())
+    {
+        GetSink(context)->diagnose(subDirectiveToken, Diagnostics::pragmaOnceIgnored);
```

Comment:
For the record: In the long term, I don't think this check is one we want to be making, but it seems to match the way we are detecting things being the "same" file for `#pragma once` handling today. If our same-ness metric changes down the line (and it realistically should), then we should get rid of this error condition.